ci/infra/testrunner/README.md

```diff
@@ -1,63 +1,213 @@
+username: "mysuserid"
 ```
 
-or
+### Platform
+
+There are some arguments that are currently at the top level of the configuration, but are actually related to the platform:
+
+- log_dir: path to the directory where platform logs are collected. Defaults to `<workspace>/testrunner_logs` 
+- nodeuser: the userid used to login into the platform nodes. Optional. 
+- ssh_option: specifies the location of the key used to access nodes. Can have two possible values:
+  - "id_shared": uses the key located at `<skuba src directory>/ci/infra/id_shared` (default)
+  - "id_rsa": uses the user's key located at `$HOME/.ssh/id_rsa`
 
-Copy `testrunner/vars.yaml` to `testrunner/myvars.yaml` set the variables in the there and use the `--vars myvars.yaml` arg
+#### Terraform
 
+General setting for terraform-based platforms such as [Openstack](#openstack) and [VMware](#vmware). 
+
+* internal_net: name of the network used when provisioning the platform. Defaults to `stack_name`
+* mirror: URL for the repository mirrors to be used when setting up the skuba nodes, replacing the URL of the repositories defined in terraform. Used, for instance to switch to development repositories or internal repositories when running in the CI pipeline.
+* plugin_dir: directory used for retrieving terraform plugins. If not set, plugins are installed using terraform [discovery mechanism](https://www.terraform.io/docs/extend/how-terraform-works.html#discovery)
+* retries: maximum number of attempts to recover from failures during terraform provisioning 
+* stack name: the unique name of the platform stack on the shared infrastructure, used as prefix by many resources such as networks, nodes, among others. If not specified, the `username` is used.
+* tfdir: path to the terraform files. Testrunner must have writing permissions to this directory. Defaults to `skuba.srcpath/ci/infra`.
+* tfvars: name of the terraform variables file to be used. Defaults to "terraform.tfvars.json.ci.example"
+
+Example
 ```
-workspace: "" # Working directory for testrunner
-...
 terraform:
-  stack_name: "" # name of the stack
+  stack_name="my-test-stack"
+```
+
+#### Openstack
+
+* openrc: path to the environment setup script
+
+Example:
 ```
+openstack:
+  openrc: "/home/myuser/my-openrc.sh"
+```
+
+#### VMware
 
-You can also set things like where the skuba binary and src are
+* env_file: path to environment variables file
+
+Example:
 ```
-SKUBA_BINPATH=/path/to/bin/skuba
-SKUBA_SRCPATH=/path/to/src/skuba
+vmware: 
+   env_file: "/path/to/env/file"
 ```
 
-or
+### Skuba
+
+The Skuba section defines the location and execution options for the `skuba` command. As `testrunner` can be used either from a local development or testing environment or a CI pipeline, the configuration allows to define the location of both the source and the binary. Please notice that the source location is used as default location for other configuration elements, such as terraform files, even if the `skuba` binary is specified. 
```

Comment:
```suggestion
The Skuba section defines the location and execution options for the `skuba` command. As `testrunner` can be used either from a CI pipeline or a local development or testing environment, the configuration allows defining the location of both the source and the binary. Please notice that the source location is used as the default location for other configuration elements, such as terraform files, even if the `skuba` binary is specified. 
```
Put single-part before the multi-part to clarify the "or", and a couple of minor grammatical tweaks